control-plane-operator/hostedclusterconfigoperator/controllers/machine/machine.go

```diff
@@ -82,6 +85,24 @@ func (r *reconciler) reconcileKubevirtDefaultIngressEndpoints(ctx context.Contex
 		return fmt.Errorf("missing default ingress passthrough Service %s/%s ports", cpService.Namespace, cpService.Name)
 	}
 
+	machine := &capiv1.Machine{}
+	if err := r.client.Get(ctx, client.ObjectKey(machineKey), machine); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("not found", "Machine", r)
+			machine = nil
+		}
+		return fmt.Errorf("failed to get Machine: %w", err)
+	}
+
+	if machine == nil {
+		return r.kubevirtInfraClient.DeleteAllOf(ctx, &discoveryv1.EndpointSlice{},
+			client.InNamespace(cpService.Namespace),
+			client.MatchingLabels{
+				"kubernetes.io/service-name": cpService.Name,
+				"hypershift.io/machine-name": machineKey.Name,
+			})
+	}
```

Comment:
this would only be reliable if we put a finalizer on the machine, which i don't think we should do in this controller.